### PR TITLE
Clean up and serializers and add examples

### PR DESCRIPTION
**Pull request by michaelrkn**
Clean up formatting and add examples of how to customize serializers.

### CODE REVIEW

File: source/models/customizing-serializers.md [hunk 1]
```diff
@@ -135,7 +136,40 @@ import DS from 'ember-data';
...
-#### ID
+To change the format of the data that is sent to the backend store, you can use
+the `serialize` hook:
+
+```app/serializers/application.js
+import DS from 'ember-data';
+
+export default DS.JSONSerializer.extend({
+  serialize(snapshot, options) {
+    // help! what do i do here?
```
Comment (michaelrkn) [inline source/models/customizing-serializers.md]: Help! What do I do here? Can someone provide a code example similar to the one below for normalizing?

**Comment by michaelrkn** [comment]
@wecc and/or @bmac, could you review this to make sure it's accurate, and help me with my partial example?

File: source/models/customizing-serializers.md [hunk 2]
```diff
@@ -1,23 +1,24 @@
-## Serializers
+In Ember Data, serializers format the data sent to and recieved from
```
Comment (locks) [inline source/models/customizing-serializers.md]: `received`

File: source/models/customizing-serializers.md [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
+the backend store. By default, Ember Data serializes data using the
+[JSON API](http://jsonapi.org/) format. If your backend uses a different
+format, Ember Data allows you to customize the serializer or use a
+different serializer entirely.
 
-In Ember Data the Serializer is a class that is responsible for
-transforming the data returned by your adapter into a JSON API document
-that Ember Data can understand. Serializers can be customized to work
-with any backend. By default, Ember Data ships with 3 Serializers. The
+Ember Data ships with 3 Serializers. The
```
Comment (locks) [inline source/models/customizing-serializers.md]: which?

File: source/models/customizing-serializers.md (hunk 1 above)
Comment (locks) [inline source/models/customizing-serializers.md]: @bmac @wecc

Comment (wecc) [inline source/models/customizing-serializers.md]: Something like this?

```js
  serialize: function(snapshot, options) {
    var json = this._super(...arguments);

    /*
     * In this example our server expects the attributes `amount` and `currency`
     * to be embedded in `cost`.
     * 
     * Before:
     *   { id: 1, name: 'My Product', amount: 100, currency: 'SEK' }
     * 
     * After:
     *   { id: 1, name: 'My Product', cost: { amount: 100, currency: 'SEK' } }
     */

    json.cost = {
      amount: json.amount,
      currency: json.currency
    };

    delete json.amount;
    delete json.currency;

    return json;
  },
```

**Comment by wecc** [comment]
There are a couple of instances of `"id": 123` (outside of your changes), they should be `"id": "123"` as IDs in JSON API are strings.

File: source/models/customizing-serializers.md [hunk 4]
```diff
@@ -135,7 +136,40 @@ import DS from 'ember-data';
...
+```
+
+If your backend store provides data in a format other than JSON API, you can
+use the `normalizeResponse` hook to change the data into the JSON API format:
+
+```app/serializers/application.js
+import DS from 'ember-data';
+
+export default DS.JSONSerializer.extend({
+  normalizeResponse(store, primaryModelClass, payload, id, requestType) {
```
Comment (wecc) [inline source/models/customizing-serializers.md]: We should really encourage people to use `this._super`.

Basically:
Before you call super, the payload is in the format that the serializer expects. Munge it but keep that format.
After you call super, the payload is now in JSON API format, feel free to munge as long as you keep it JSON API.

Delegating conversion to super gives you A LOT for free + you have the flexibility to make customizations in two different formats.

Comment (michaelrkn) [inline source/models/customizing-serializers.md]: :+1:

So like this?

```
  normalizeResponse: function(store, primaryModelClass, payload, id, requestType) {
    /*
     * In this example our server provides the attributes `amount` and `currency`
     * embedded in `cost`.
     * 
     * Before:
     *   { id: 1, name: 'My Product', cost: { amount: 100, currency: 'SEK' } }
     * 
     * After:
     *   { id: 1, name: 'My Product', amount: 100, currency: 'SEK' }
     */

    payload.amount = payload.cost.amount;
    payload.amount = payload.cost.currency;

    delete payload.cost;

    return this._super(store, primaryModelClass, payload, id, requestType);
  },
```

Comment (wecc) [inline source/models/customizing-serializers.md]: LGTM!

Comment (wecc) [inline source/models/customizing-serializers.md]: Oh wait, no. `normalizeResponse` is for the full payload, so it kind of depends.

Comment (wecc) [inline source/models/customizing-serializers.md]: If you want to manipulate the data for a single resource then the `normalize` hook is better.

Comment (wecc) [inline source/models/customizing-serializers.md]: You can use `normalizeResponse` to remove top-level data for server-responses. Like if your server kind of complies to JSONSerializer but puts everything in a `{ "response": ... }`, you can get do `payload = payload.response`

Comment (michaelrkn) [inline source/models/customizing-serializers.md]: So if you use an application serializer, use `normalizeResponse`, and if you use a model serializer, then use `normalize`?

Comment (wecc) [inline source/models/customizing-serializers.md]: I'd say that's a pretty common pattern, yes.

**Comment by michaelrkn** [comment]
@wecc Can you do one last review? It took me a couple days to find the time to verify all the code examples worked properly.

**Pull request merged**